packages/core/src/types/overrides/react.ts

```diff
@@ -0,0 +1,83 @@
+import type { RefineComponentName } from ".";
+
+export type RefineReactComponent<Name extends RefineComponentName> =
+  React.ComponentType<RefineReactComponentProps<Name>>;
+
+/**
+ * Here we define all of the props that each component takes. This is used both
+ * in our library code for our default components and in the overrides end
+ * users can provide.
+ */
+export type RefineReactComponentPropsMap = {
+  group: {
+    className: string;
+  };
+  "group.addCriterionButton": {
+    className: string;
+  };
```

Comment:
For example if we wanted this component to have an `onClick` prop we would define that here like so:

```suggestion
  "group.addCriterionButton": {
    className: string;
    onClick: () => void
  };
```

Then in our library code we reuse that type to ensure everything syncs up and is type-safe.